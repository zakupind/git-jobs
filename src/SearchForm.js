import React from 'react';

import { connect } from 'react-redux';

import { searchSubmit,
         searchNull } from './REDUX/action';

class Search extends React.Component {
    constructor(props) {
        super(props);

    this.state = {
        description: '',
        location: '',
        fullTime: false
        }
    }

    handleChange(event) {
        const { name, value } = event.target;
        this.setState({[name]: value});
    }

    submit(event) {
        event.preventDefault();
        if (this.state.description === '' && this.state.location === '') {
            this.props.searchNull();
        } else {
            this.props.searchSubmit(this.state);
        }
    }

    render() {
        return(
            <div className="search-form-wrapper">
                <form className="search-form"
                      onSubmit={(event) => this.submit(event)}>
                    <input className="search-form__query"
                           placeholder="Поисковый запрос"
                           name="description"
                           value={this.description}
                           onChange={(event) => this.handleChange(event)}/>
                    
                    <div className="search-form__location_and_fulltime">
                        <input className="search-form__location"
                               placeholder="Город"
                               name="location"
                               value={this.location}
                               onChange={(event) => this.handleChange(event)}/>

                        <div className="check-fulltime">
                            <input className="check-fulltime__check"
                                   type="checkbox"
                                   name="fullTime"
                                   onChange={(event) => this.handleChange(event)}
                                   defaultChecked={this.fullTime}/>

                            <label>Только полный день</label>
                        </div>
                    </div>

                    <button className="search__button">Найти</button>
                </form>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    description: state.search.description,
    location: state.search.location,
    fullTime: state.search.fullTime
})

const mapDispatchToProps = {
    searchSubmit,
    searchNull
}

export default connect(mapStateToProps, mapDispatchToProps)(Search);