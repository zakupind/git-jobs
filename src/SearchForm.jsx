import React from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';

import {
  searchSubmit,
  searchNull,
} from './store/action';

class Search extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: '',
      location: '',
      fullTime: false,
    };
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    // const { fullTime: fullTimeDes } = this.state;
    if (name === 'fullTime') {
      this.setState(({ fullTimeDes }) => ({ fullTime: !fullTimeDes }));
    } else {
      this.setState({ [name]: value });
    }
  }

  submit = (event) => {
    const { description, location } = this.state;
    const { searchNull: searchNullAction, searchSubmit: searchSubmitAction } = this.props;

    event.preventDefault();
    if (description === '' && location === '') {
      searchNullAction();
    } else {
      searchSubmitAction(this.state);
    }
  }

  render() {
    return (
      <div className="search-form-wrapper">
        <form
          className="search-form"
          onSubmit={this.submit}
        >
          <input
            className="search-form__query"
            placeholder="Поисковый запрос"
            name="description"
            value={this.description}
            onChange={this.handleChange}
          />

          <div className="search-form__location_and_fulltime">
            <input
              className="search-form__location"
              placeholder="Город"
              name="location"
              value={this.location}
              onChange={this.handleChange}
            />

            <div className="check-fulltime">

              <label htmlFor="fullTime">
                <input
                  className="check-fulltime__check"
                  type="checkbox"
                  name="fullTime"
                  onChange={this.handleChange}
                  defaultChecked={this.fullTime}
                />
                Только полный день
              </label>
            </div>
          </div>

          <button type="submit" className="search__button">Найти</button>
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = {
  searchSubmit,
  searchNull,
};

Search.propTypes = {
  searchSubmit: PropTypes.func.isRequired,
  searchNull: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(Search);
