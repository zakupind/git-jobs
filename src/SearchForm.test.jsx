import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import { Search } from './SearchForm';

enzyme.configure({ adapter: new Adapter() });

describe('тест формы поиска', () => {
  let props;
  let component;

  describe('общие тесты', () => {
    beforeEach(() => {
      props = {
        searchSubmit: () => {},
        searchNull: () => {},
      };

      component = mount(<Search {...props} />);
    });

    test('snapshot search', () => {
      const tree = renderer
        .create(<Search {...props} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('ввод данных', () => {
    props = {
      searchSubmit: () => {},
      searchNull: () => {},
    };

    component = mount(<Search {...props} />);

    test('input description test', () => {
      const description = component.find('input').at(0);
      description.simulate('change',
        { target: { value: 'python' } });

      expect(component.state('description')).toEqual('python');
    });

    test('input location test', () => {
      const location = component.find('input').at(1);
      location.simulate('change',
        { target: { value: 'new-york' } });

      expect(component.state('location')).toEqual('new-york');
    });

    test('input fullTime click', () => {
      const fullTime = component.find('input').at(2);
      fullTime.simulate('click');

      expect(component.state('fullTime')).toEqual(true);
    });

    // test('button click', () => {
    //   const submit = component.find('button');
    //   submit.simulate('click');

    //   expect(searchSubmit)
    // });
  });
});
