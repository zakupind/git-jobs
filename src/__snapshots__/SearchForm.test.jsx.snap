// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тест формы поиска общие тесты snapshot search 1`] = `
<div
  className="search-form-wrapper"
>
  <form
    className="search-form"
    onSubmit={[Function]}
  >
    <input
      className="search-form__query"
      name="description"
      onChange={[Function]}
      placeholder="Поисковый запрос"
    />
    <div
      className="search-form__location_and_fulltime"
    >
      <input
        className="search-form__location"
        name="location"
        onChange={[Function]}
        placeholder="Город"
      />
      <div
        className="check-fulltime"
      >
        <label
          htmlFor="fullTime"
        >
          <input
            className="check-fulltime__check"
            name="fullTime"
            onClick={[Function]}
            type="checkbox"
          />
          Только полный день
        </label>
      </div>
    </div>
    <button
      className="search__button"
      type="submit"
    >
      Найти
    </button>
  </form>
</div>
`;
